benchmarks/driver.py

```diff
@@ -0,0 +1,333 @@
+                 "iters" : "1000",
+                 "flags": ""
+             },
+         },
+         "128x256x512":  {
+             ...
+         },
+     ]
+
+    The compiler paths/includes/libraries will be detected automatically.
+    The source file will be compiled together with the common driver in the
+    root of the benchmark directory, so they should only contain the kernel
+    logic to be called by the driver.
+
+    The arguments to the MLIR harness will be detected automatically.
+"""
+
+import os
+import sys
+import re
+import shlex
+import argparse
+import json
+
+sys.path.append('harness')
+
+from Logger import Logger
+from Execute import Execute
+from TPPHelper import TPPHelper
+
+class Environment(object):
+    def __init__(self, args, logger):
+        helper = TPPHelper(logger)
+        self.base_dir = os.path.realpath(os.path.dirname(__file__))
+        self.root_dir = helper.findGitRoot(".")
+        programs = helper.findTPPProgs(self.root_dir)
+        for _, path in programs.items():
+            if os.path.exists(path):
+                self.bin_dir = os.path.realpath(os.path.dirname(path))
+                parent = os.path.join(self.bin_dir, os.path.pardir)
+                self.build_dir = os.path.realpath(parent)
+                self.lib_dir = os.path.join(self.build_dir, "lib")
+                self.inc_path = os.path.join(self.root_dir, "include")
+                self.lib_inc_path = os.path.join(self.build_dir, "_deps", "xsmm-src", "include")
+                break
+        assert(self.build_dir)
+        self.bench_dir = os.path.join(self.root_dir, "benchmarks")
+        self.harness = os.path.join(self.bench_dir, "harness", "controller.py")
+        self.test_dir = os.path.join(self.root_dir, "test", "Benchmarks")
+        # Pass arguments down to benchmarks, if known
+        self.extra_args = list()
+        if args.verbose > 0:
+            for v in range(args.verbose - args.quiet):
+                self.extra_args.append("-v")
+
+class BaseRun(object):
+    """ Base class for all runs """
+
+    def __init__(self, name, env, json, logger):
+        self.logger = logger
+        self.name = name
+        self.env = env
+        self.source = json["source"]
+        self.iters = json["iters"]
+        self.flags = json["flags"]
+        self.runner = Execute(logger)
+        self.stdout = ""
+        self.stderr = ""
+
+    def compile(self):
+        # This is optional
+        return True
+
+    def run(self):
+        # This is mandatory
+        return False
+
+    def cleanup(self):
+        # This is optional
+        return True
+
+class CPPRun(BaseRun):
+    """ C++ runs """
+
+    def __init__(self, name, env, json, logger):
+        BaseRun.__init__(self, name, env, json, logger)
+        assert(json["type"] == "C++")
+        source_dir = os.path.dirname(self.source)
+        self.binary = os.path.join(source_dir, name)
```

Comment:
I'd suggest adding some prefix/suffix to the created binaries and adding it to `.gitignore` to avoid accidental commits with binaries when `-k` flag is used.